### PR TITLE
Add order summary and top food items

### PR DESCRIPTION
**Pull request by jelneo**

### CODE REVIEW

File: controllers/staff.js [hunk 1]
```diff
@@ -1,6 +1,58 @@
...
+    SELECT DISTINCT rid, oid
+    FROM Collates natural join Orders
+)
+SELECT count(distinct oid) as numOrders, sum(finalPrice) as totalCost, DATE_PART('month', orderTime) as mth, DATE_PART('year', orderTime) as yr
+FROM RestaurantOrderPair natural join Orders
+WHERE rid = $1 AND deliveredTime IS NOT NULL
+GROUP BY mth, yr;
+` });
+
+const psGetTop5Food = new PS({ name: 'get-top-5-food', text:
```
Comment (MuhdNurKamal) [inline controllers/staff.js]: Would be good to put top 'number' as a parameter instead of hardcoding 5

File: routes/staff.js [hunk 2]
```diff
@@ -1,6 +1,62 @@
...
+            orderStatistics = await getTotalOrdersAndCost(req.params.rid);
+        } catch (err) {
+            return next(err)
+        }
+        return res.send(orderStatistics);
+    });
+
+    router.get("/min-max-date", async (req, res, next) => {
+        let dates = [];
+        console.log
```
Comment (MuhdNurKamal) [inline routes/staff.js]: A wild console.log appeared

File: controllers/staff.js [hunk 3]
```diff
@@ -1,6 +1,58 @@
...
+const getTotalOrdersAndCost = async (rid) => {
+    return await db.any(psGetTotalOrdersAndCost, [rid]);
+};
+
+const getTop5Food = async (rid, month, year) => {
+    return await db.any(psGetTop5Food, [rid, month, year]);
+};
+
+const getMinMaxDate = async () => {
+    return await db.one(psGetMinMaxDate);
```
Comment (MuhdNurKamal) [inline controllers/staff.js]: Use oneOrNone to deal with empty table

File: controllers/staff.js (hunk 1 above)
Comment (MuhdNurKamal) [reply controllers/staff.js]: actually i dont see any sorting done in the sql. is it rly a top query

File: controllers/staff.js [hunk 4]
```diff
@@ -1,6 +1,58 @@
...
+WITH FoodFreq AS (
+    SELECT fname, sum(qty) as totalQty, DATE_PART('month', orderTime) as mth, DATE_PART('year', orderTime) as yr
+    FROM Collates natural join Orders
+    WHERE rid = $1 
+    GROUP BY fname, mth, yr
+)
+SELECT fname, totalQty
+FROM FoodFreq
+WHERE mth = $2 AND yr = $3;
+ `});
```
Comment (AdityaA1998) [inline controllers/staff.js]: Are you trying to gather monthly data, or overall? plus i don't see any sorting logic here

File: db/fillTables.js [hunk 5]
```diff
@@ -251,6 +272,28 @@ async function addFrequents(arr) {
...
+        await db.none(
+            `Insert into Orders (oid, riderId, customerId, orderTime, deliveredTime, deliveryFee, isDeliveryFeeWaived, departForR, arriveAtR, departFromR, finalPrice, addrId, pid) Values
+            (${arr[0]}, ${arr[1]}, ${arr[2]}, '${arr[3]}', '${arr[4]}', ${arr[5]}, ${arr[6]}, '${arr[7]}', '${arr[8]}', '${arr[9]}', ${arr[10]}, ${arr[11]}, ${arr[12]})`
+        );
+    } catch (error) {
+        console.log(error, 'Failed to add orders');
+    }
+}
+
 async function deleteTables() {
```
Comment (AdityaA1998) [inline db/fillTables.js]: Can delete orders and collates tables as well.

**Pull request merged**